app/utilities/metadata_parser.py

```diff
@@ -41,10 +43,27 @@ class DateString(fields.DateTime):
     """Currently, runner cannot handle Date objects in metadata
     Since all metadata is serialized and deserialized to JSON.
     This custom field deserializes Dates to strings.
+
+    :param out_format: Output date string format. If falsy, the output
+        format is the same as the input.
     """
 
+    def __init__(self, *args, out_format: str = None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._out_format = out_format or kwargs.get("format")
+
     def _deserialize(self, *args, **kwargs):  # pylint: disable=arguments-differ
-        return super()._deserialize(*args, **kwargs).strftime("%Y-%m-%d")
+        date = super()._deserialize(*args, **kwargs).strftime(self._out_format)
+
+        if "%z" in self._out_format:
+            dt, tz = date.split("+")
+            # Add a colon separating the timezone offset hours and minutes
+            if ":" not in date.split("+")[-1]:
+                new_tz = f"{tz[:2]}:{tz[2:]}"
+
+            date = f"{dt}+{new_tz}"
+
+        return date
```

Comment:
I don't think most of this is needed. Marshmallow has support for ISO 8601 already. Replacing this with the below work as expected for me and the new tests also pass.

```py
class DateString(fields.DateTime):
    """Currently, runner cannot handle Date objects in metadata
    Since all metadata is serialized and deserialized to JSON.
    This custom field deserializes Dates to strings.
    """

    def _deserialize(self, *args, **kwargs):  # pylint: disable=arguments-differ
        date = super()._deserialize(*args, **kwargs)

        if self.format == "iso8601":
            return date.isoformat()

        return date.strftime(self.format)

VALIDATORS = {
    "date": functools.partial(DateString, format="%Y-%m-%d", required=True),
    "uuid": functools.partial(UUIDString, required=True),
    "boolean": functools.partial(fields.Boolean, required=True),
    "string": functools.partial(fields.String, required=True),
    "url": functools.partial(fields.Url, required=True),
    "iso_8601_date": functools.partial(DateString, format="iso8601", required=True),
}
```

- The format arg isn't needed as that is the default, but I included it for clarity. 
- Format can be `iso` or `iso8601`, used `iso8601` to be explicit.
- I think the name should be `iso_8601_date` or `_date_string` for clarity, as it is not a datetime object.

Reply:
Funny that, I tried all sorts of combinations and it just kept on returning not quite the right format. ta